tests/utils/test_text_util.py

```diff
@@ -159,6 +159,33 @@ def test_hash_list(self):
         self.assertEqual(result_from_list, hash_from_string.digest())
         self.assertEqual(hash_from_string.hexdigest(), '6367c48dd193d56ea7b0baad25b19455e529f5ee')
 
+    def test_empty_strings(self):
+        self.assertTrue(textutil.is_str_none_or_whitespace(' '))
+        self.assertTrue(textutil.is_str_none_or_whitespace('\t'))
+        self.assertTrue(textutil.is_str_none_or_whitespace('\n'))
+        self.assertTrue(textutil.is_str_none_or_whitespace(' \t'))
+        self.assertTrue(textutil.is_str_none_or_whitespace(' \r\n'))
+
+        self.assertTrue(textutil.is_str_empty(' '))
+        self.assertTrue(textutil.is_str_empty('\t'))
+        self.assertTrue(textutil.is_str_empty('\n'))
+        self.assertTrue(textutil.is_str_empty(' \t'))
+        self.assertTrue(textutil.is_str_empty(' \r\n'))
+
+        self.assertFalse(textutil.is_str_none_or_whitespace(u' \x01 '))
+        self.assertFalse(textutil.is_str_none_or_whitespace(u'foo'))
+        self.assertFalse(textutil.is_str_none_or_whitespace('bar'))
+
+        self.assertFalse(textutil.is_str_empty(u' \x01 '))
+        self.assertFalse(textutil.is_str_empty(u'foo'))
+        self.assertFalse(textutil.is_str_empty('bar'))
+
+        self.assertFalse(textutil.is_str_none_or_whitespace(u'\x00'))
+        self.assertFalse(textutil.is_str_none_or_whitespace(u' \x00 '))
+
+        self.assertTrue(textutil.is_str_empty(u'\x00'))
+        self.assertTrue(textutil.is_str_empty(u' \x00 '))
```

Comment:
These last two cases are critical: they identify scenarios where the two methods are _required_ to return different values. Consider reframing those test cases to assert that the distinction exists. That is...
```
hex_null = u'\x00'
hex_null_with_blanks = u' \x00 '
self.assertTrue(textutil.is_str_empty(hex_null) and not textutil.is_str_none_or_whitespace(hex_null))
self.assertTrue(textutil.is_str_empty(hex_null_with_blanks) and not textutil.is_str_none_or_whitespace(hex_null_with_blanks))
```

Reply:
Updated